resolwe_bio/extensions.py

```diff
@@ -1,9 +1,65 @@
 """Resolwe-bio extensions of Resolwe core."""
+from django.db.models import Q
 from django_filters import rest_framework as filters
 
 from resolwe.composer import composer
 
 
+class ExtendedCollectionFilter:
+    """Collection filter extension."""
+
+    descriptor__general__species = filters.CharFilter(
```

Comment:
You have to return collections that contain at least one sample (instance of Entity class) that has a descriptor with the given species.

So something like this:

```
descriptor__general__species = filters.CharFilter(
        field_name="entity__descriptor__general__species", lookup_expr="contains", distinct=True
    )
```

The distinct is necessary or collection with multiple samples matching given criteria will be returned multiple times.


All other queries also have to be modified.